File: Chapter 03/C#/Section_3_3_3_LSP/After/Midfielder.cs
using System;

namespace Chapter03.LSP.After;

/// <summary>
/// LSP SOLUTION: True Substitutability.
/// 
/// ARCHITECTURE NOTE: Just like the Forward, the Midfielder fully supports the 
/// behavior expected of a generic Player. We can swap a Forward for a Midfielder 
/// at runtime, and the application will remain perfectly stable.
/// </summary>
public class Midfielder : Player 
{
    /// <summary>
    /// Flawlessly fulfills the base class contract.
    /// </summary>
    public override void PlayFieldPosition() 
    {
        Console.WriteLine("  [Midfielder] Controlling the midfield, passing and tackling."); 
    } 
}

File: Chapter 03/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter03/section_3_3_3_lsp/after/Midfielder.java
package com.grokkingsoftwarearchitecture.chapter03.section_3_3_3_lsp.after;

/**
 * LSP SOLUTION: True Substitutability.
 * * ARCHITECTURE NOTE: Just like the Forward, the Midfielder fully supports the 
 * behavior expected of a generic Player. We can swap a Forward for a Midfielder 
 * at runtime, and the application will remain perfectly stable.
 */
public class Midfielder extends Player {
    
    /**
     * Flawlessly fulfills the base class contract.
     */
    @Override
    public void playFieldPosition() {
        System.out.println("  [Midfielder] Controlling the midfield, passing and tackling.");
    }
}